### PR TITLE
fix: download wheels in parallel to avoid deadlock

### PR DESCRIPTION
**Pull request by baszalmstra**
This paralellizes the download/building of wheels (vs concurrent) to avoid a deadlock issue we encountered.

### CODE REVIEW

File: src/install_pypi.rs [hunk 1]
```diff
@@ -323,7 +356,7 @@ fn stream_python_artifacts<'a>(
             }
         })
         // TODO: put this back on 20 when there is not deadlock anymore.
```
Comment (tdejager) [inline src/install_pypi.rs:358]: This can be removed :)

**Comment by ruben-arts** [comment]
Test with:
```
cd examples/pypi
rm ~/.cache/rattler/cache/pypi/ .pixi -r && pixi i 
```

**Review by tdejager — APPROVED** [review]
LGTM! Ill try making a PR later to remove all WheelBuilder lifetimes so this becomes a bit easier as well.

File: src/install_pypi.rs (hunk 1 above)
Comment (baszalmstra) [reply src/install_pypi.rs:358]:
```suggestion
```

**Review by ruben-arts — APPROVED** [review]
:rocket:

**Pull request merged**